### PR TITLE
F#104 change name current pose

### LINKED ISSUE #104: get_current_pose method of robot_commander is badly named
It should be called something like "get_movegroup_state"

Changing it will involve grepping a at least all of sr_interface to find places where the old style method is used. It's a relatively recent function so it's probably not used outside sr_interface anywhere yet.

### PR DESCRIPTION
**Pull request by beatrizleon**
Fixes #104 

Renamed the current get_current_pose to get_current_state and replaced where it was used.

Created a new get_current_pose to actually get the pose of the end effector. It takes the reference frame as an argument. If it is given, it uses tf to find the pose if not it uses the default get_current_pose from moveit which uses the planning_frame as reference (/world)

### CODE REVIEW

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py [hunk 1]
```diff
@@ -213,13 +216,38 @@ def get_named_target_joint_values(self, name):
 
         return output
 
-    def get_current_pose(self):
+    def get_current_pose(self, reference_frame=None):
```
Comment (ugocupcic) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py:219]: add a docstring to both functions to clarify - especially the @return type

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py [hunk 2]
```diff
@@ -213,13 +216,38 @@ def get_named_target_joint_values(self, name):
...
+                current_pose.orientation.w = rot[3]
+                return current_pose
+            except (tf.Exception, tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException):
+                rospy.logwarn("Couldn't get the pose from " + self._move_group_commander.get_end_effector_link() +
+                              " in " + reference_frame + " reference frame")
+            return None
+        else:
+            return self._move_group_commander.get_current_pose().pose
+
+    def get_current_state(self):
```
Comment (ugocupcic) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py:250]: here too

File: sr_example/scripts/sr_example/hand_examples/sr_right_hand_follow_finger_trajectory.py [hunk 3]
```diff
@@ -59,26 +55,30 @@
...
+waypoints.append(pose2)
+
+pose3 = geometry_msgs.msg.Pose()
+pose3.position.x = 0.033
+pose3.position.y = -0.068
+pose3.position.z = 0.151
+pose3.orientation.w = 1.00
+waypoints.append(pose3)
+
+waypoints.append(pose2)
```
Comment (toliver) [inline sr_example/scripts/sr_example/hand_examples/sr_right_hand_follow_finger_trajectory.py:79]: Why are we appending them in reverse order?

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py (hunk 1 above)
Comment (toliver) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py:219]: Could you add documentation to the method?

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py [hunk 4]
```diff
@@ -213,13 +216,38 @@ def get_named_target_joint_values(self, name):
...
         joint_names = self._move_group_commander._g.get_active_joints()
         joint_values = self._move_group_commander._g.get_current_joint_values()
 
         return dict(zip(joint_names, joint_values))
 
-    def get_current_pose_bounded(self):
+    def get_current_state_bounded(self):
```
Comment (toliver) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py:260]: Doc

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py [hunk 5]
```diff
@@ -213,13 +216,38 @@ def get_named_target_joint_values(self, name):
 
         return output
 
-    def get_current_pose(self):
+    def get_current_pose(self, reference_frame=None):
+        current_pose = geometry_msgs.msg.Pose()
```
Comment (toliver) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py]: This assignation can be inside the `try`.

File: sr_example/scripts/sr_example/hand_examples/sr_right_hand_follow_finger_trajectory.py (hunk 3 above)
Comment (beatrizleon) [inline sr_example/scripts/sr_example/hand_examples/sr_right_hand_follow_finger_trajectory.py:79]: I created a trajectory first moving horizontally (abduction/adduction) of the finger and then moving vertically (flexing the finger) so to do the latter I appended three poses and then, to move back to the initial pose, I appended them but in reverse order.

Comment (toliver) [inline sr_example/scripts/sr_example/hand_examples/sr_right_hand_follow_finger_trajectory.py:79]: Sorry I didn't see the first 3 appends. :)

File: sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py [hunk 6]
```diff
@@ -213,13 +216,52 @@ def get_named_target_joint_values(self, name):
...
+        """
         joint_names = self._move_group_commander._g.get_active_joints()
         joint_values = self._move_group_commander._g.get_current_joint_values()
 
         return dict(zip(joint_names, joint_values))
 
-    def get_current_pose_bounded(self):
+    def get_current_state_bounded(self):
+        """
+        Get the current joint state of the group being used, checking that they are within each joint limits.
```
Comment (toliver) [inline sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py]: A bit more accurate to say `enforcing` rather than checking.

**Pull request merged**